google/yaml_validator.rb

```diff
@@ -100,7 +100,8 @@ def check_type(name, object, type)
       elsif type.is_a? ::Array
         return if type.find_index(:boolean) && [TrueClass, FalseClass].find_index(object.class)
         return unless type.find_index(object.class).nil?
-      elsif object.is_a?(type)
+      # check if class is or inherits from type
```

Comment:
Instead of adding this check, what do you think of adding both as top-level fields? I think it leads to a bit more natural of a template- I'll expand below.

Reply:
I think leaving them as inherited still makes sense because both classes will still have several properties shared between them; primarily timeouts but I can see properties like `clear_id_on_error` or similar being another shared attribute. There will also be other places such as the website generation that only want to know if a resource is async and not care about the async implementation under the hood.

Reply:
👍